### ISSUE TITLE
extract algorithm from plugin

### ISSUE THREAD

**Issue by pfzero** [miromannino/Justified-Gallery#135]
Hi, is there an easy solution to extract the algorithm from the code? Something like a module which takes an options object and an array of images and gives as output the photos to be displayed on each row.

I am asking this as I want to use the plugin but want to get rid of jQuery, by using for example [ReactJS](https://facebook.github.io/react/).

Thanks.

**Comment by miromannino** [comment]
Hello, the exact algorithm for justification is not that complicated. The problem is all the other things, the async loading of images (to know their sizes), the responsiveness, etc...
Tell me what are the exact problem you are facing, because maybe I could reduce the amount of used jQuery (e.g. using jqlite, because I would like to do the integration with angular).

**Comment by pfzero** [comment]
Hi @miromannino, I recently published a repo on [github](https://github.com/pfzero/justifiedGallery) and also published on npm. I use the idea from the justified gallery (from flickr blogs) but to be honest didn;t have the patience to look into your code and extract it from there.

The main idea of this question was that to make the plugin view framework agnostic so it can be easily used with jQuery or others. Because, for example, the React like web components libraries are becoming trendy and it would be handy to have some core module (containing the algorithm) which can be then easily extended to be used with any library.

So I thought that we can work together for the core module and maybe create widgets for the different libraries. What do you think?

**Comment by miromannino** [comment]
Yes, of course that is a good Idea. We can of course collaborate.

But the only problem I see is that maybe you are redoing something again. In this way you will spend a lot of time to do it again. Instead, could be better to spend much less time to port all the JG to a vanillaJS version, removing all jQuery stuff. I've started to do that, so will be easier from the current version. I did a separate class to manage the gallery, and the jQuery plugin uses that class. I will do an AngularJS directive that will only use that class, for example, without considering if that is a jQuery plugin. The only problem now is that in this moment that class is using jQuery stuff (so I only need to remove all the $(el).children(); in el.children for example). The complicated part where I asynchronously prefetch the images, for example, doesn't use jQuery anymore, but a simple vanillaJS Image object.

Let me know, but starting again from zero will require a lot of time. For the basic things it's simple (I did the first version of JG in days), but then having a very stable and professional object, with infinite scroll capabilities, with spinner loader, different thumbnail sizes, etc... requires much more time. 

I think we could better join our strength instead of "fork it again" (i.e. "linux style"), dispersing energies, and going nowhere :)
